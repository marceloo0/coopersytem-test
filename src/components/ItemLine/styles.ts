import styled from 'styled-components/native';

export const Container = styled.View`
  background-color: #fff;
`;
export const Content = styled.View`
  flex-direction: row;
  justify-content: space-between;
  border-bottom-color: #e8eded;
  border-bottom-width: 1px;
  margin-left: 20px;
  padding: 10px 20px 10px 0;
`;
export const Title = styled.Text`
  font-size: 21px;
  font-weight: bold;
`;
export const Description = styled.Text`
  font-size: 21px;
  font-weight: bold;
  color: #cecece;
`;
