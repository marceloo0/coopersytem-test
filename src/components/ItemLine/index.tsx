import React from 'react';

import {Container, Content, Title, Description} from './styles';

interface ItemLineProps {
  title: string;
  description: string;
}

const ItemLine: React.FC<ItemLineProps> = ({title, description}) => {
  return (
    <Container>
      <Content>
        <Title>{title}</Title>
        <Description>{description}</Description>
      </Content>
    </Container>
  );
};

export default ItemLine;
