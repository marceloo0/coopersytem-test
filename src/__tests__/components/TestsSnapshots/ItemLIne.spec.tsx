import React from 'react';
import renderer from 'react-test-renderer';
import ItemLine from '../../../components/ItemLine';

test('renders correctly', () => {
  const itemLine = renderer
    .create(
      <ItemLine title="Saldo total disponivel" description="R$ 9.000,00" />,
    )
    .toJSON();
  expect(itemLine).toMatchSnapshot();
});
