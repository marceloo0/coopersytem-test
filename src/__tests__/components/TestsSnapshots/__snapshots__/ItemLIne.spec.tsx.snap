// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "borderBottomColor": "#e8eded",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginLeft": 20,
          "paddingBottom": 10,
          "paddingLeft": 0,
          "paddingRight": 20,
          "paddingTop": 10,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 21,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Saldo total disponivel
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#cecece",
            "fontSize": 21,
            "fontWeight": "bold",
          },
        ]
      }
    >
      R$ 9.000,00
    </Text>
  </View>
</View>
`;
